velox/substrait/SubstraitToVeloxPlan.cpp

```diff
@@ -19,6 +19,112 @@
 
 namespace facebook::velox::substrait {
 
+std::shared_ptr<const core::PlanNode> SubstraitVeloxPlanConverter::toVeloxPlan(
+    const ::substrait::JoinRel& sJoin) {
+  if (!sJoin.has_left()) {
+    VELOX_FAIL("Left Rel is expected in JoinRel.");
+  }
+  if (!sJoin.has_right()) {
+    VELOX_FAIL("Right Rel is expected in JoinRel.");
+  }
+
+  auto leftNode = toVeloxPlan(sJoin.left());
+  auto rightNode = toVeloxPlan(sJoin.right());
+
+  auto outputSize =
+      leftNode->outputType()->size() + rightNode->outputType()->size();
+  std::vector<std::string> outputNames;
+  std::vector<std::shared_ptr<const Type>> outputTypes;
+  outputNames.reserve(outputSize);
+  outputTypes.reserve(outputSize);
+  for (const auto& node : {leftNode, rightNode}) {
+    auto names = node->outputType()->names();
```

Comment:
Can the output names be set here directly and omit the project node after the join node?

Reply:
There's a restriction that the output names of join should be consistent with its children's output names. https://github.com/oap-project/velox/blob/cbd85fd0c77e193664ebad3005d28c9448324f2e/velox/core/PlanNode.cpp#L282-L299

Reply:
I see. There is similar restriction for the agg grouping columns. But the name change seems to be still not needed, because the next plan node will find its input names through the input row type.

Reply:
I get it. Removed the extra projection.